### DISCUSSION TITLE
How to set focus on an input element from an element outside the scope?

### DISCUSSION THREAD

**Discussion by sanjayojha** [alpinejs/alpine#3152] [1. Help] [answered]
Hi, I am using latest version of Alpine. I need to set focus on the input searchbox as soon as it become visible. Here is my code for reference.
 ```html
<div x-data="{search: false}">
    <button x-click="search = true">Click to open search box</button>
    <div x-data="searchwidget" x-show="search">
          <input type="search" x-model="term" x-ref="searchInput" />
    </div>
</div>
```

Data handler for search input
```javascript
document.addEventListener('alpine:init', () => {
          Alpine.data('searchwidget', () => ({
               term: "",
          }));
});
```
I tried to put `$refs.searchInput.focus()` on button element, but its not working.

**Comment by ekwoka** [comment]
That's not really outside the scope.

It's still inside a parent.

What I would recommend is using the `@alpinejs/focus` core plugin.

https://alpinejs.dev/plugins/focus

It comes with a lot of helpers for managing focus.

**Reply by sanjayojha** [reply]
Yes, its inside a parent. Is there any other method doing it without plugin?

**Reply by ekwoka** [reply]
I'd probably have the search widget over the entire popover instead of nested inside. The thing is that refs get added to the nearest component, not all parent components.

But the focus plugin is THE way to do it as it's just better and handles a lot of issues for you, ntm you likely have real need for it regardless in an app beyond the most limited scale.

**Comment by SimoTod** [comment, marked as answer]
Alternatively, you can use x-if and x-init (It won't work with x-show because x-init only runs when the component is mounted. x-show just toggles the display property while x-if removes and reinserts the HTML every time).

```html
<div x-data="{search: false}">
  <button @click="search = !search">Click to open/close search box</button>
  <template x-if="search">
    <div x-data="searchwidget">
      <input type="search" x-model="term" x-init="$el.focus()" />
    </div>
  </template>
</div>

<script>
  document.addEventListener('alpine:init', () => {
    Alpine.data('searchwidget', () => ({
      term: "",
    }));
  });
</script>

<script src="https://unpkg.com/alpinejs" defer></script>
```

**Reply by sanjayojha** [reply]
Thanks. This looks neat solution. Will try to implement it.
